application-development/helloworld-springboot-auth/README.md

```diff
@@ -0,0 +1,106 @@
+
+[maven]: https://maven.apache.org
+[maven-download]: https://maven.apache.org/download.cgi
+[maven-install]: https://maven.apache.org/install.html
+
+### Create a Project in the Google Cloud Platform Console
+
+If you haven't already created a project, create one now. Projects enable you to
+manage all Google Cloud Platform resources for your app, including deployment,
+access control, billing, and services.
+
+1. Open the [Cloud Platform Console][cloud-console].
+1. In the drop-down menu at the top, select **Create a project**.
+1. Give your project a name.
+1. Make a note of the project ID, which might be different from the project
+   name. The project ID is used in commands and in configurations.
+
+[cloud-console]: https://console.cloud.google.com/
+
+### Enable billing for your project.
+
+If you haven't already enabled billing for your project, [enable
+billing][enable-billing] now.  Enabling billing allows the application to
+consume billable resources such as running instances and storing data.
+
+[enable-billing]: https://console.cloud.google.com/project/_/settings
+
+### Install the Google Cloud SDK.
+
+If you haven't already installed the Google Cloud SDK, [install and initialize
+the Google Cloud SDK][cloud-sdk] now. The SDK contains tools and libraries that
+enable you to create and manage resources on Google Cloud Platform.
+
+[cloud-sdk]: https://cloud.google.com/sdk/
+
+### Install the Google App Engine SDK for Java
+
+
+```
+gcloud components update app-engine-java
+gcloud components update
+```
+
+### Configure the `app.yaml` descriptor
+
+The [`app.yaml`][app-yaml] descriptor is used to describe URL
+dispatch and resource requirements.  This example sets
+[`manual_scaling`][manual-scaling] to 1 to minimize possible costs.
+Dont change the manual scaling to anything other than 1 , unleess you refactoir to code to have a shared session store.
+These settings should be revisited for production use.
+
+[app-yaml]: https://cloud.google.com/appengine/docs/flexible/java/configuring-your-app-with-app-yaml
+[manual-scaling]: https://cloud.google.com/appengine/docs/flexible/java/configuring-your-app-with-app-yaml#manual-scaling
+[oauth-steps]: https://cloud.google.com/java/getting-started/authenticate-users
+
+## Change the OAuth client settings
+Follow the instructions as described [here][oauth-steps]
+
+Also add `http://[YOUR_PROJECT_ID].appspot.com/login` and `https://[YOUR_PROJECT_ID].appspot.com/login` to the Authorized redirect URIs 
+
+Change the clientId and clientSecret in `src/main/resources/application.yml` 
+
+## Run the application locally
+
+1. Set the correct Cloud SDK project via `gcloud config set project
+   YOUR_PROJECT` to the ID of your application.
+1. Run `mvn spring-boot:run`
+1. Visit http://localhost:8080
+
+
+## Deploy to App Engine flexible environment
+
+1. `mvn appengine:deploy`
+1. Visit `http://YOUR_PROJECT.appspot.com`.
+* if this is the first deployment, you need to run `gcloud app create`
```

Comment:
Yes, the first deployment.  The error that mvn gives is hidden and hard to find.  Simple change to avoid a bad experience :)